File: apps/server/src/university/work_positions/dtos.rs
use serde::{Deserialize, Serialize};
use validator::Validate;

#[derive(Debug, Serialize, Deserialize, Validate)]
pub struct CreateAcademicWorkPositionDto {
    #[validate(length(
        min = 1,
        max = 255,
        message = "El nombre debe tener entre 1 y 255 caracteres"
    ))]
    pub name: String,
}

#[derive(Debug, Serialize, Deserialize, Validate, Default)]
pub struct GetWorkPositionsQuery {
    #[validate(length(
        min = 1,
        max = 255,
        message = "El nombre debe tener entre 1 y 255 caracteres"
    ))]
    pub name: Option<String>,
}

File: apps/client/src/lib/university/work-positions/dtos.ts
import * as v from "valibot"

export interface AcademicWorkPosition {
	id: string
	name: string
}

export const createPositionSchema = v.object({
	name: v.pipe(
		v.string(),
		v.minLength(1, "El nombre debe tener entre 1 y 255 caracteres"),
		v.maxLength(255, "El nombre debe tener entre 1 y 255 caracteres"),
	),
})

export type CreatePositionDto = v.InferInput<typeof createPositionSchema>
